include/time_units.h

```diff
@@ -0,0 +1,860 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_TIME_UNITS_H_
+#define ZEPHYR_INCLUDE_TIME_UNITS_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Exhaustively enumerated, highly optimized time unit conversion API */
+
+#if defined(CONFIG_TIMER_READS_ITS_FREQUENCY_AT_RUNTIME)
+__syscall int z_clock_hw_cycles_per_sec_runtime_get(void);
+
+static inline int z_impl_z_clock_hw_cycles_per_sec_runtime_get(void)
+{
+	extern int z_clock_hw_cycles_per_sec;
+
+	return z_clock_hw_cycles_per_sec;
+}
+#endif /* CONFIG_TIMER_READS_ITS_FREQUENCY_AT_RUNTIME */
+
+static inline int sys_clock_hw_cycles_per_sec(void)
+{
+#if defined(CONFIG_TIMER_READS_ITS_FREQUENCY_AT_RUNTIME)
```

Comment:
It is, this is just a cut-and-paste of the older code from sys_clock.h, which is why it's unchanged.  Can fix, sure.

Reply:
Actually no, sorry.  The expression would work, but the function being selected isn't visible if the kconfig isn't set.  I could make that work some other way, but honestly I don't think it would be any more readable.

FWIW: the only reason that mess exists is that historically the HPET driver (only that driver) has read its clock frequency at initialization time and hasn't had a compile time HZ value.  So it was just a variable in memory instead, but then userspace came around and user code needed to get to the variable that was in kernel memory, and so it needed to be a system call.